#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 4
  coordinateSystem "NUE"
  gpsCoordinateSystem "WGS84"
  gpsReference 135 35 0
}
Viewpoint {
  orientation 0.5320471236481618 0.8456396548009424 0.04265480565962596 6.094139777946267
  position -0.35831770355689335 1.0928591689940368 2.364581197774435
  follow "biwako-x"
  followType "None"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF SWIMMING_POOL SolidPipe {
  translation 0 0.25 0
  height 1.5
  radius 500
  subdivision 144
  appearance DEF DARK_BROWN PBRAppearance {
    baseColor 0.501961 0.337255 0.2
    roughness 1.1102230246251565e-16
    metalness 0
  }
}
DEF STILL_WATER Fluid {
  translation 0 0.2 0
  children [
    DEF S Shape {
      appearance DEF WATER_APPEARANCE PBRAppearance {
        baseColor 0 0.501961 1
        transparency 0.5
        roughness 1.1102230246251565e-16
        metalness 0
      }
      geometry Cylinder {
        height 1
        radius 499
        subdivision 12
      }
    }
  ]
  name "swimming pool"
  boundingObject USE S
  locked TRUE
}
DEF SWIMMING_POOL_BOTTOM Transform {
  translation 0 0.02 0
  children [
    Shape {
      appearance USE DARK_BROWN
      geometry Cylinder {
        height 0.02
        radius 499
        subdivision 24
      }
    }
  ]
}
Floor {
}
DEF BIWAKO-X Robot {
  translation 0 0.58 0
  rotation 0 1 0 -0.785395307179586
  children [
    DEF BODY Solid {
      children [
        DEF BODY Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Cylinder {
            height 0.4
            radius 0.05
            subdivision 16
          }
        }
      ]
      name "body"
      boundingObject USE BODY
      physics Physics {
        density -1
        mass 1.5
        centerOfMass [
          0 -0.05 0
        ]
      }
    }
    DEF TH1 Transform {
      translation 0.13 0.08 0
      rotation 1 9.58979e-09 1.88117e-06 -1.5707953071795862
      children [
        Propeller {
          shaftAxis 0 1 0
          thrustConstants 0.7 0.1
          torqueConstants 0 0
          device DEF MOTOR1 RotationalMotor {
            name "thruster1"
            consumptionFactor 1
            maxVelocity 50
            maxTorque 1
          }
          fastHelix Solid {
            children [
              DEF PROP Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  metalness 0
                }
                geometry Cylinder {
                  height 0.075
                  radius 0.01
                }
              }
            ]
          }
          slowHelix Solid {
            children [
              USE PROP
            ]
          }
        }
      ]
    }
    DEF TH4 Transform {
      translation -0.13 0.08 0
      rotation -4.6356318433936376e-07 0.707106281186096 0.7071072811864936 3.14159
      children [
        Propeller {
          shaftAxis 0 1 0
          thrustConstants 0.7 0.1
          torqueConstants 0 0
          device DEF MOTOR4 RotationalMotor {
            name "thruster4"
            maxVelocity 50
            maxTorque 1
          }
          fastHelix Solid {
            children [
              DEF PROP Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  metalness 0
                }
                geometry Cylinder {
                  height 0.075
                  radius 0.01
                }
              }
            ]
          }
          slowHelix Solid {
            rotation 0 0 1 3.14159
            children [
              USE PROP
            ]
          }
        }
      ]
    }
    DEF TH3 Transform {
      translation 0 0.08 0.13
      rotation 0.577349935856137 -0.577349935856137 -0.5773509358560258 -2.0943853071795866
      children [
        Propeller {
          shaftAxis 0 1 0
          thrustConstants 0.7 0.1
          torqueConstants 0 0
          device DEF MOOTOR3 RotationalMotor {
            name "thruster3"
            consumptionFactor 0
            maxVelocity 50
            maxTorque 1
          }
          fastHelix Solid {
            children [
              DEF PROP Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  metalness 0
                }
                geometry Cylinder {
                  height 0.075
                  radius 0.01
                }
              }
            ]
          }
          slowHelix Solid {
            children [
              USE PROP
            ]
          }
        }
      ]
    }
    DEF TH2 Transform {
      translation 0 0.08 -0.13
      rotation 0.5773479358471216 0.5773479358471216 0.5773549358463438 -2.094395307179586
      children [
        Propeller {
          shaftAxis 0 1 0
          thrustConstants 0.7 0.1
          torqueConstants 0 0
          device DEF MOTOR2 RotationalMotor {
            name "thruster2"
            consumptionFactor 0
            maxVelocity 50
            maxTorque 1
          }
          fastHelix Solid {
            children [
              DEF PROP Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  metalness 0
                }
                geometry Cylinder {
                  height 0.075
                  radius 0.01
                }
              }
            ]
          }
          slowHelix Solid {
            children [
              USE PROP
            ]
          }
        }
      ]
    }
    DEF BUOY Solid {
      translation 0.27 0.12 0
      rotation 0 0 1 -1.5707953071795862
      children [
        DEF S_BUOY Shape {
          appearance PBRAppearance {
            baseColor 1 0.9 0
            metalness 0
          }
          geometry Cylinder {
            height 0.1
            radius 0.035
          }
        }
      ]
      name "buoy"
      immersionProperties [
        DEF SWIMMING_POOL_IMMERSION_PROPERTIES ImmersionProperties {
          fluidName "swimming pool"
          dragForceCoefficients 0.1 0 0
          dragTorqueCoefficients 0.001 0 0
          viscousResistanceTorqueCoefficient 0.005
        }
      ]
      boundingObject USE S_BUOY
      physics DEF P_BUOY Physics {
        density -1
        mass 0.027
      }
    }
    DEF BUOY Solid {
      translation -0.27 0.12 0
      rotation 0 0 1 -1.5707953071795862
      children [
        DEF S_BUOY Shape {
          appearance PBRAppearance {
            baseColor 1 0.9 0
            metalness 0
          }
          geometry Cylinder {
            height 0.1
            radius 0.035
          }
        }
      ]
      name "buoy(1)"
      immersionProperties [
        DEF SWIMMING_POOL_IMMERSION_PROPERTIES ImmersionProperties {
          fluidName "swimming pool"
          dragForceCoefficients 0.1 0 0
          dragTorqueCoefficients 0.001 0 0
          viscousResistanceTorqueCoefficient 0.005
        }
      ]
      boundingObject USE S_BUOY
      physics DEF P_BUOY Physics {
        density -1
        mass 0.027
      }
    }
    DEF BUOY Solid {
      translation 0 0.12 0.27
      rotation 0.5773509358560258 -0.577349935856137 -0.577349935856137 2.09439
      children [
        DEF S_BUOY Shape {
          appearance PBRAppearance {
            baseColor 1 0.9 0
            metalness 0
          }
          geometry Cylinder {
            height 0.1
            radius 0.035
          }
        }
      ]
      name "buoy(2)"
      immersionProperties [
        DEF SWIMMING_POOL_IMMERSION_PROPERTIES ImmersionProperties {
          fluidName "swimming pool"
          dragForceCoefficients 0.1 0 0
          dragTorqueCoefficients 0.001 0 0
          viscousResistanceTorqueCoefficient 0.005
        }
      ]
      boundingObject USE S_BUOY
      physics DEF P_BUOY Physics {
        density -1
        mass 0.027
      }
    }
    DEF BUOY Solid {
      translation 0 0.12 -0.27
      rotation 0.5773509358560258 -0.577349935856137 -0.577349935856137 2.09439
      children [
        DEF S_BUOY Shape {
          appearance PBRAppearance {
            baseColor 1 0.9 0
            metalness 0
          }
          geometry Cylinder {
            height 0.1
            radius 0.035
          }
        }
      ]
      name "buoy(3)"
      immersionProperties [
        DEF SWIMMING_POOL_IMMERSION_PROPERTIES ImmersionProperties {
          fluidName "swimming pool"
          dragForceCoefficients 0.1 0 0
          dragTorqueCoefficients 0.001 0 0
          viscousResistanceTorqueCoefficient 0.005
        }
      ]
      boundingObject USE S_BUOY
      physics DEF P_BUOY Physics {
        density -1
        mass 0.027
      }
    }
    DEF TH_BAR Solid {
      translation 0 0.12 0.2
      children [
        DEF S_BAR Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.015 0.015 0.3
          }
        }
      ]
    }
    DEF TH_BAR Solid {
      translation 0 0.12 -0.2
      children [
        DEF S_BAR Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.015 0.015 0.3
          }
        }
      ]
      name "solid(1)"
    }
    DEF TH_BAR Solid {
      translation 0.2 0.12 0
      rotation 0 1 0 -1.5707953071795862
      children [
        DEF S_BAR Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.015 0.015 0.3
          }
        }
      ]
      name "solid(2)"
    }
    DEF TH_BAR Solid {
      translation -0.2 0.12 0
      rotation 0 1 0 -1.5707953071795862
      children [
        DEF S_BAR Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.015 0.015 0.3
          }
        }
      ]
      name "solid(3)"
    }
    DEF Sensor Solid {
      children [
        Compass {
          translation 0 0.2 0
          rotation 9.589809999941977e-09 -0.9999999999939495 3.4786599999789525e-06 -0.785395307179586
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.01
              }
            }
          ]
        }
        GPS {
          translation 0 0.2 0
          rotation 0 1 0 0.785398
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.01
              }
            }
          ]
        }
      ]
      name "Sensor"
    }
    DEF HEAD Solid {
      translation 0.02 0.2 -0.02
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0.2
            metalness 0
          }
          geometry Cylinder {
            height 0.001
            radius 0.01
          }
        }
      ]
      name "head"
    }
  ]
  name "biwako-x"
  immersionProperties [
    DEF SWIMMING_POOL_IMMERSION_PROPERTIES ImmersionProperties {
      fluidName "swimming pool"
      dragForceCoefficients 0.1 0 0
      dragTorqueCoefficients 0.001 0 0
      viscousResistanceTorqueCoefficient 0.005
    }
  ]
  boundingObject Group {
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 2.51
    centerOfMass [
      0 -0.05 0
    ]
    inertiaMatrix [
      0.04385139745635754 0.003926990816987243 0.04385139745635754
      0 0 0
    ]
    damping Damping {
    }
  }
  controller "Python_BIWAKO-X"
  battery [
    0
  ]
}
